src/components/modals/ManagePlanDerivationGroupsModal.svelte

```diff
@@ -0,0 +1,414 @@
+  );
+
+  $: filteredDerivationGroups = $derivationGroups.filter(derivationGroup => {
+    const filterTextLowerCase = filterText.toLowerCase();
+    const includesName = derivationGroup.name.toLocaleLowerCase().includes(filterTextLowerCase);
+    return includesName;
+  });
+
+  $: {
+    columnDefs = [
+      {
+        field: 'name',
+        filter: 'string',
+        headerName: 'Derivation Group',
+        resizable: true,
+        sortable: true,
+        suppressAutoSize: false,
+        suppressSizeToFit: false,
+      },
+      {
+        field: 'source_type_name',
+        filter: 'string',
+        headerName: 'Source type',
+        resizable: true,
+        sortable: true,
+        suppressAutoSize: false,
+        suppressSizeToFit: false,
+      },
+      {
+        field: 'derived_event_total',
+        filter: 'number',
+        headerName: 'Derived Events in Derivation Group',
+        sortable: true,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        valueFormatter: params => {
+          return params?.value.length;
+        },
+        width: 250,
+      },
+      {
+        cellDataType: 'boolean',
+        editable: true,
+        headerName: 'Included in Plan',
+        resizable: false,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        valueGetter: (params: ValueGetterParams<DerivationGroup>) => {
+          const { data } = params;
+          if (data) {
+            return $selectedPlanDerivationGroupNames.includes(data.name);
+          }
+          return false;
+        },
+        width: 115,
+      },
+      {
+        cellClass: 'action-cell-container',
+        cellRenderer: (params: DerivationGroupCellRendererParams) => {
+          const actionsDiv = document.createElement('div');
+          actionsDiv.className = 'actions-cell';
+          new DataGridActions({
+            props: {
+              rowData: params.data,
+              viewCallback: params.viewDerivationGroup,
+              viewTooltip: {
+                content: 'View Derivation Group',
+                placement: 'bottom',
+              },
+            },
+            target: actionsDiv,
+          });
+
+          return actionsDiv;
+        },
+        cellRendererParams: {
+          viewDerivationGroup,
+        } as CellRendererParams,
+        headerName: '',
+        resizable: false,
+        sortable: false,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        width: 40,
+      },
+    ];
+  }
+
+  function viewDerivationGroup(derivationGroup: DerivationGroup) {
+    const dg = $derivationGroups.find(dg => dg.name === derivationGroup.name);
+    selectedDerivationGroup = dg;
+  }
+
+  function getRowId(derivationGroup: DerivationGroup): string {
+    return `${derivationGroup.name}:${derivationGroup.source_type_name}`;
+  }
+
+  function handleUpdateLayerNewDerivationGroups(newLayers: Layer[]) {
```

Comment:
I think that this modal is doing too much outside of its scope. IMO a modal should be in charge of presenting some information and triggering a change relevant to the information being displayed. I think this modal should only handle inserting/deleting derivation groups and not handle anything timeline related. This could cause some unintended clashes with components more devoted to managing the timeline or view. Can this logic be moved somewhere more relevant? Is it even necessary to update the timeline explicitly after the effect function calls? If I were to trigger these changes, would someone else viewing the same timeline not get these changes?

Reply:
Agree it's doing a lot, but not exactly sure where to move it - some context:
- The purpose of this modal is to allow the user to link/unlink (or insert/delete) derivation groups to plans, with also a secondary purpose of letting the user investigate/view the derivation group to see more information (and lead them to decide if they want to insert/delete it)
- This function & the related code is part of a fix we did related to reactivity not working properly when derivation groups were associated with a plan and the **external event types** that the events in the derivation group used needed to be added to the filter. Using the logic the timeline editor uses as reference, we refactored the logic here so that when the effect is called, any external event layers that exist in the timeline(s) + row(s) will be re-generated to have the proper event type filter (one that contains the 'new' external event types used by the derivation group)

With that said, I _think_ we need to update the timeline explicitly after the effect function calls, but let me know if I'm misunderstanding! I suppose maybe the function itself could be moved out to another file, but I think we'd still want to call it after the modal calls the effect. To your last point, I think the optimal outcome for someone else viewing the timeline would be to echo the layer updates to them as well, so that if User A adds a derivation group to Plan A, and User B is also looking at Plan A, they see the update in real-time.

Reply:
I would say that the general convention that we have right now is that the `effects` file is what updates the necessary stores if necessary. That way the update logic is centralized in case another component needs to invoke the same functions and needs to update the relevant stores as well.

I think I might have misunderstood how this worked. So you're saying that when a user makes changes to the derivations linked to a plan via this modal, if they refresh the page, they would have to redo the linking/unlinking they made previously to get the same timeline?

Reply:
That makes sense!

And not quite - so the expected function (and what currently happens) is that when the user links or unlinks a derivation group in the modal, that is a **permanent** change that should automatically reflect on the timeline. So if you're in Plan A and you link 'Example Derivation Group', you'll instantly see those events pop-up on your external events row (which also requires adding whatever event type they belong to to the row's filter, which was the reason for re-generating the layer objects). Additionally, that link is there until a user **explicitly unlinks** 'Example Derivation Group'.

I hope that context makes sense - with that said though I think it makes sense and is definitely possible to move this function doing the actual logic of re-creating layers to `effects.ts`!

Reply:
Nice! Okay that's what I was hoping it would do. So what I'm getting at is that I think there already exists some logic that sets up the timeline to correctly display the events on external data load, which means that you should be able to rely on that logic being triggered to update the timeline again since the data is being updated via the socket when the user links/unlinks them through this modal. I think updating it in the modal or effect after the request to update succeeds might just be causing the timeline to update twice. I'm open for a huddle to help clarify this if needed since I'm terrible at explaining things.

Reply:
I think I'm following and I think one thing I didn't fully realize before was that when we _first_ associate a Derivation Group, the event types inside are not only added as possibilities to each External Event Layer's filter, but **selected** as active filters (and thus should instantly start showing the events on the timeline). I don't believe that's how Activity Directives work unless I'm missing it, since it seems like you need to explicitly turn an Activity Type filter onto show it on a row (except the Activities by Type row since it auto-fills & selects all your types on page load).

With that said, looking into how Activity filters work a bit I think we might be able to re-use `viewAddFilterToRow` and `viewAddFilterItemsToRow` to achieve what we want - queuing those off of a successful derivation group linking. Am I understanding all that correctly, and does that fulfill what you're looking for? If not we can definitely hop on a huddle, I'm still trying to wrap my head around all the UI intricacies so it's way more than possible I'm missing something obvious

Reply:
I thiiiiink that's what I'm alluding to. Basically, the UI state should be being derived from the db data as much as possible. Any updates to the data should be propagated from the db via websockets to the UI instead of the UI state being directly manipulated (unless it's a transient state). I'm not familiar enough with the view filters yet to say whether or not that's the correct place, you might know more about it at this point. We could wait for @AaronPlave's input to see how best to handle that if you like.

Reply:
I'm definitely on-board with what you're saying now conceptually, and think I have a few (uncommitted) changes that'll fulfill what you're looking for! Sounds like a good plan to wait til' Aaron is back, then maybe the 3 of us can tag-up, look at these changes and discuss where to go next with it. Thanks for the help + patience!

Reply:
Chatted w. Bryan and Aaron about this - summarizing some notes and steps below:

- Logic as is makes sense and does (currently) need to exist to allow plans w. no prior derivation group/external event association(s) to "begin" to incorporate them when derivation groups are associated. Future work by Aaron may alter the need/logic here, but as-is this needs to stay
- To close out this comment, @JosephVolosin will do one of two things:

1. Keep the current logic but move pieces out to `effects.ts` so that the "effects" are not occurring within the modal. The filter/layer logic can be incorporated into the existing `insertDerivationGroupForPlan` effect

2. Investigate (and possibly implement) if External Events can be incorporated into the `Activity & Resource Types` menu in the plan view. This would basically get rid of the need for the above step (1) as then users would be able to drag & drop event types from associated derivation groups to update filters, create new rows, etc. This is the *preferred* route if it isn't a major hassle to implement

Reply:
Ended up going with the second option which I think makes the most sense, and allows us to remove all the excess from that modal. So now the expected actions for a user are: 1) Link a derivation group to the plan, 2) Open the 'Activity, Resource, External Event Type' menu item, 3) Add filter to, or create a new, row. See [this commit](https://github.com/NASA-AMMOS/aerie-ui/commit/c56fc3f283487fc27ae3bf9a9610882154d7622d) for the work.

One note, there's some styling issues now from adding the extra tab to the menu item, and also expanding the menu item's name to include external events... thoughts on how we want to fix this?
![image](https://github.com/user-attachments/assets/a97d88f4-a595-4479-8e94-945c35a8d77e)

Reply:
Fixed in a different thread I believe.